torch/sparse/_triton_ops_meta.py

```diff
@@ -0,0 +1,1607 @@
+        (8192, 8192, 8192, 128, 128): (4, 1, 4),
+        (8192, 8192, 16384, 16, 16): (4, 2, 1),
+        (8192, 8192, 16384, 32, 32): (4, 4, 1),
+        (8192, 8192, 16384, 64, 64): (4, 3, 2),
+        (8192, 8192, 16384, 128, 128): (4, 1, 4),
+        (8192, 8192, 32768, 16, 16): (4, 2, 1),
+        (8192, 8192, 32768, 32, 32): (4, 4, 1),
+        (8192, 8192, 32768, 64, 64): (4, 3, 2),
+        (8192, 8192, 32768, 128, 128): (4, 1, 4),
+        (8192, 8192, 65536, 16, 16): (4, 2, 1),
+        (8192, 8192, 65536, 32, 32): (4, 3, 1),
+        (8192, 8192, 65536, 64, 64): (4, 3, 2),
+        (8192, 8192, 65536, 128, 128): (4, 1, 4),
+        (8192, 8192, 131072, 16, 16): (4, 1, 4),
+        (8192, 8192, 131072, 32, 32): (4, 2, 1),
+        (8192, 8192, 131072, 64, 64): (4, 3, 2),
+        (8192, 8192, 131072, 128, 128): (4, 1, 4),
+        (16384, 16384, 256, 16, 16): (2, 7, 1),
+        (16384, 16384, 256, 32, 32): (4, 4, 2),
+        (16384, 16384, 256, 64, 64): (4, 4, 4),
+        (16384, 16384, 256, 128, 128): (4, 2, 8),
+        (16384, 16384, 512, 16, 16): (4, 3, 1),
+        (16384, 16384, 512, 32, 32): (4, 5, 2),
+        (16384, 16384, 512, 64, 64): (4, 4, 4),
+        (16384, 16384, 512, 128, 128): (4, 2, 8),
+        (16384, 16384, 1024, 16, 16): (4, 3, 1),
+        (16384, 16384, 1024, 32, 32): (4, 4, 1),
+        (16384, 16384, 1024, 64, 64): (4, 3, 2),
+        (16384, 16384, 1024, 128, 128): (4, 1, 4),
+        (16384, 16384, 2048, 16, 16): (4, 3, 1),
+        (16384, 16384, 2048, 32, 32): (4, 4, 1),
+        (16384, 16384, 2048, 64, 64): (4, 4, 4),
+        (16384, 16384, 2048, 128, 128): (4, 1, 4),
+        (16384, 16384, 4096, 16, 16): (4, 3, 1),
+        (16384, 16384, 4096, 32, 32): (4, 4, 1),
+        (16384, 16384, 4096, 64, 64): (4, 3, 2),
+        (16384, 16384, 4096, 128, 128): (4, 1, 4),
+        (16384, 16384, 8192, 16, 16): (2, 2, 1),
+        (16384, 16384, 8192, 32, 32): (4, 4, 1),
+        (16384, 16384, 8192, 64, 64): (4, 3, 2),
+        (16384, 16384, 8192, 128, 128): (4, 1, 4),
+        (16384, 16384, 16384, 16, 16): (4, 2, 1),
+        (16384, 16384, 16384, 32, 32): (4, 4, 1),
+        (16384, 16384, 16384, 64, 64): (4, 3, 2),
+        (16384, 16384, 16384, 128, 128): (4, 1, 4),
+        (16384, 16384, 32768, 16, 16): (2, 2, 1),
+        (16384, 16384, 32768, 32, 32): (4, 5, 1),
+        (16384, 16384, 32768, 64, 64): (4, 3, 2),
+        (16384, 16384, 32768, 128, 128): (4, 1, 4),
+        (16384, 16384, 65536, 16, 16): (4, 2, 1),
+        (16384, 16384, 65536, 32, 32): (4, 4, 1),
+        (16384, 16384, 65536, 64, 64): (4, 3, 2),
+        (16384, 16384, 65536, 128, 128): (4, 1, 4),
+        (16384, 16384, 131072, 16, 16): (4, 1, 4),
+        (16384, 16384, 131072, 32, 32): (4, 2, 1),
+        (16384, 16384, 131072, 64, 64): (4, 3, 2),
+        (16384, 16384, 131072, 128, 128): (4, 1, 4),
+    },
+    ("scatter_mm", "NVIDIA A100-SXM4-80GB", (0, torch.float16, 0.5)): {
```

Comment:
I think the GPU type key might be slightly too constrained, since I'd expect this to also work on the 40GB version for example.

Reply:
Right. There is actually a number of different A100 models (at least 14 according to https://github.com/NVIDIA/open-gpu-kernel-modules) that may have the same set of optimal triton kernel meta parameters.

My last commit now maps all NVIDIA A100 models, that are not in the triton_ops_meta.py database, to "NVIDIA A100-SXM4-80GB" so that the device matching constraint will be less strict.

The only way to know if this assumption is reasonable, is to compute the optimal parameters for a particular A100 model and compare with the existing set. This can be done in a follow-up when we get access to such GPU models.